starter-code/js/index.js

```diff
@@ -23,22 +17,77 @@ function updateSubtot($product) { // $product = tr (row) element of product
 
 function calcAll() {
   var total = 0;
+  var $products = document.querySelectorAll(".product"); //need to be INSIDE the function because each time we click on button, the function will find all the products (even the ones added with the create button)
   $products.forEach($product => {
     updateSubtot($product);
     total += Number($product.querySelector(".subtot span").textContent);
   });
   document.querySelector(".total span").textContent = total;
 }
+
+var $calc = document.getElementById('calc');
 $calc.onclick = calcAll; // calcAll is called ONLY when .calc (=Calculate prices button) is clicked
 
 //**********************************************************************
 
-// Iteration 4
+// Iteration 4 - deleting a product
+
+function deleteProduct(event) { // event = the parameter = the click
+  //console.log(event.currentTarget); // it's the clicked-button (button is the currentTarget)
+  const td = event.currentTarget.parentNode; // the parent node of our event.currentTarget (AKA the button) is the <td> cell
+  const tr = td.parentNode; // the parent node of the <td> is the <tr> row (THAT we want to REMOVE)
+  const tbody = tr.parentNode; // the parent node of the <tr> row is the <tbody> !
+  tbody.removeChild(tr); // so to remove a row we select the tbody and then remove its child : the tr row !
+  console.log("deleteProduct() is called");
+  calcAll(); // update the total price without the removed product
+}
+
+function initDeleteButtons() {
+  var $deleteButtons = document.querySelectorAll(".btn.btn-delete");
+  $deleteButtons.forEach($deleteButton => {
+    $deleteButton.onclick = deleteProduct;
+  });
+}
+
+initDeleteButtons();
+
+//**********************************************************************
+
+// Iteration 5 - creating a new product
+
+//1) assign click event to the "create" button and by clicking call a function createNewProductRow:
+var $createButton = document.getElementById("create");
+$createButton.onclick = createNewProductRow;
+
+//2) that function gets the values of input "product name" and "price unit" in new variables:
+function createNewProductRow() {
+  var newProductName = document.querySelector("#newProduct").value;
+  var newPrice = document.querySelector("#newPrice ").value;
+
+  //3) add a row in the tbody table with innerHTML(***) by writing all the details of the tr (".product") within backticks ``and replace new values of product and price unit.
+  // (***: with a += !! otherwise I will erase/replace all the content of tbody by the new product)
+  var $cart = document.querySelector('#cart tbody');
+  $cart.innerHTML += `
+    <tr class="product">
+      <td class="name">
+        <span>${newProductName}</span>
+      </td>
+
+      <td class="pu">$<span>${newPrice}</span></td>
+
+      <td class="qty">
+        <label>
+          <input type="number" placeholder="enter a quantity" min="0" />
+        </label>
+      </td>
```

Comment:
When you create your elements and append them to the innerHTML the values in the inputs disappear, to fix that behaviour you could create a tr element and then append it to the cart body.
`  var $cart = document.querySelector("#cart tbody");
  const tr = document.createElement("tr");
  tr.className ="product";
  tr.innerHTML = `
    <td class="name">
      <span>${newProductName}</span>
    </td>

    <td class="pu">$<span>${newPrice}</span></td>

    <td class="qty">
      <label>
        <input type="number" placeholder="enter a quantity" min="0" />
      </label>
    </td>

    <td class="subtot">$<span>0</span></td>

    <td class="rm">
      <button class="btn btn-delete">Delete</button>
    </td>
`;
  $cart.appendChild(tr);`